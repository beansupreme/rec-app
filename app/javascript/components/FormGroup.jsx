import React from "react"

class FormGroup extends React.Component {
  render() {
    return (
      <div className="form-group">
        <label htmlFor={this.props.inputId}>{this.props.labelText}</label>
        <input id={this.props.inputId} className="form-control" type={this.props.type} name={this.props.formName}/>
      </div>
    )
  }
}

export default FormGroup;