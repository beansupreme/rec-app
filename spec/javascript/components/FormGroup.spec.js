import React, {Component} from 'react'
import {shallow, render} from 'enzyme'
import FormGroup from 'components/FormGroup'
import renderer from 'react-test-renderer'

describe('<FormGroup />', () => {
  it('renders a form group label with the label name', () => {
    const wrapper = shallow(
      <FormGroup inputId="important_field" type="text" 
      formName="custom_form[field]" labelText="Important Field" />
    );
    expect(wrapper.text()).toContain('Important Field')
  });

  it('renders as expected', () => {
    const tree = renderer.create(
      <FormGroup inputId="field_name" type="text" 
      formName="custom_form[field]" labelText="Fill this out!"/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});