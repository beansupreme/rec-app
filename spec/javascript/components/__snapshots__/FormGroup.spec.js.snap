// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormGroup /> renders as expected 1`] = `
<div
  className="form-group"
>
  <label
    htmlFor="field_name"
  >
    Fill this out!
  </label>
  <input
    className="form-control"
    id="field_name"
    name="custom_form[field]"
    type="text"
  />
</div>
`;
